modules/brushingandlinking/src/brushingandlinkingmanager.cpp

```diff
@@ -78,30 +78,46 @@ void BrushingAndLinkingManager::brush(BrushingAction action, BrushingTarget targ
     propagate(action, target);
 }
 
-bool BrushingAndLinkingManager::isModified() const { return !modifiedActions_.empty(); }
+bool BrushingAndLinkingManager::isModified() const { return !modifications_.empty(); }
 
-BrushingModifications BrushingAndLinkingManager::modifiedActions() const {
-    return modifiedActions_;
+BrushingModifications BrushingAndLinkingManager::getModifiedActions() const {
+    BrushingModifications modifiedActions(flags::empty);
+
+    for (const auto& elem : modifications_) {
+        modifiedActions |= elem.second;
+    }
+    return modifiedActions;
 }
 
-bool BrushingAndLinkingManager::modifiedFiltering(BrushingTarget target) const {
-    return modifiedActions_.contains(BrushingModification::Filtered) && modifiedTarget(target);
+bool BrushingAndLinkingManager::isFilteringModified(BrushingTarget target) const {
+    return isTargetModified(target, BrushingModification::Filtered);
 }
 
-bool BrushingAndLinkingManager::modifiedSelection(BrushingTarget target) const {
-    return modifiedActions_.contains(BrushingModification::Selected) && modifiedTarget(target);
+bool BrushingAndLinkingManager::isSelectionModified(BrushingTarget target) const {
+    return isTargetModified(target, BrushingModification::Selected);
 }
 
-bool BrushingAndLinkingManager::modifiedHighlight(BrushingTarget target) const {
-    return modifiedActions_.contains(BrushingModification::Highlighted) && modifiedTarget(target);
+bool BrushingAndLinkingManager::isHighlightModified(BrushingTarget target) const {
+    return isTargetModified(target, BrushingModification::Highlighted);
 }
 
-const std::vector<BrushingTarget>& BrushingAndLinkingManager::modifiedTargets() const {
-    return modifiedTargets_;
+std::vector<BrushingTarget> BrushingAndLinkingManager::getModifiedTargets() const {
+    return util::transform(modifications_, [](const auto& elem) { return elem.first; });
 }
 
-bool BrushingAndLinkingManager::modifiedTarget(BrushingTarget target) const {
-    return util::contains(modifiedTargets_, target);
+bool BrushingAndLinkingManager::isTargetModified(BrushingTarget target,
+                                                 BrushingAction action) const {
+    return isTargetModified(target, fromAction(action));
+}
+
+bool BrushingAndLinkingManager::isTargetModified(BrushingTarget target,
+                                                 BrushingModifications modifications) const {
+    auto it = modifications_.find(target);
+    if (it != modifications_.end()) {
+        return !(it->second | modifications).empty();
+    }
```

Comment:
Nit: you can do it it like this now :)
```suggestion
    if (auto it = modifications_.find(target); it != modifications_.end()) {
        return !(it->second | modifications).empty();
    }
```